src/constructLayoutEngine.js

```diff
@@ -63,56 +72,97 @@ export function constructLayoutEngine({
   function arrangeDomElements() {
     const path = location[resolvedRoutes.mode === "hash" ? "hash" : "pathname"];
 
-    // TODO memoize this for perf
-    const matchedRoutes = matchRoute(resolvedRoutes, path);
+    if (!path.startsWith(baseWithoutSlash)) {
+      // Base URL doesn't match, no need to recurse routes
+      return;
+    }
 
-    const initialContainer =
+    const parentContainer =
       typeof resolvedRoutes.containerEl === "string"
         ? document.querySelector(resolvedRoutes.containerEl)
         : resolvedRoutes.containerEl;
 
-    recurseRoutes(matchedRoutes.routes, initialContainer);
+    recurseRoutes({
+      path,
+      pathMatch: baseWithoutSlash,
+      routes: resolvedRoutes.routes,
+      parentContainer,
+      shouldMount: true,
+    });
+  }
+
+  function handleAppChange({ detail: { appsByNewStatus } }) {
+    appsByNewStatus.NOT_MOUNTED.concat(appsByNewStatus.NOT_LOADED).forEach(
```

Comment:
After every single-spa reroute, remove the application elements for all unmounted applications.

Reply:
Is it really necessary to add the apps that are in `NOT_LOADED` state? IIRC `NOT_LOADED` means they haven't appended dom nodes yet.

Reply:
The "new status" is NOT_LOADED. Which means that before the transition it was loaded and mounted. See https://single-spa.js.org/docs/api#custom-events